### PR TITLE
Minor: Criterion benchmarking on GKR hash proving

### PR DESCRIPTION
**Pull request by tonyfloatersu**
Running benchmark for a GKR (either vanilla or GKR^2) prover for proving a hashing under single threaded setting.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -74,22 +74,7 @@ impl Config<M31ExtConfig> {
...
-            gkr_config: M31ExtConfig,
-            gkr_scheme,
-        }
-    }
-}
-
-impl Config<BN254Config> {
-    pub fn new(gkr_scheme: GKRScheme) -> Self {
-        Config {
-            field_size: 93,
```
Comment (zhenfeizhang) [inline src/config.rs:86]: oh i realized this two shouldn't be merged. and the BN254Config isn't implemented correctly.

Comment (zhenfeizhang) [reply src/config.rs:86]:
```
            field_size: 254,
            security_bits: 100,
            #[cfg(feature = "grinding")]
            grinding_bits: 0,
```

Comment (tonyfloatersu) [reply src/config.rs:86]: yikes - I stepped on some landmine here.  I will revert my change and set proper params for `BN254Config` here

File: src/config.rs [hunk 2]
```diff
@@ -65,31 +65,16 @@ pub struct Config<C: GKRConfig> {
     pub gkr_scheme: GKRScheme,
 }
 
-impl Config<M31ExtConfig> {
+impl<C: GKRConfig> Config<C> {
     pub fn new(gkr_scheme: GKRScheme) -> Self {
         Config {
-            field_size: 93,
+            field_size: C::ChallengeField::FIELD_SIZE,
```
Comment (tonyfloatersu) [inline src/config.rs:71]: I am being lazy here and derive the field size from challenge field, which the field eventually should fall into - but feel free to debate about the slacking bits from a slightly larger field bits, I am open to changes here.

File: benches/gkr_hashes.rs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    (config, circuit)
+}
+
+fn criterion_gkr_keccak(c: &mut Criterion) {
+    let (m31_config, mut m31_circuit) =
+        benchmark_setup::<M31ExtConfig>(GKRScheme::Vanilla, KECCAK_CIRCUIT);
+    let (bn254_config, mut bn254_circuit) =
+        benchmark_setup::<BN254Config>(GKRScheme::Vanilla, KECCAK_CIRCUIT);
+
+    let mut group = c.benchmark_group("prove keccak by GKR vanilla");
```
Comment (zhenfeizhang) [inline benches/gkr_hashes.rs:32]: "prove keccak by GKR vanilla" => "single thread proving 2 keccaks by GKR vanilla"

File: benches/gkr_hashes.rs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    });
+}
+
+fn criterion_gkr_poseidon(c: &mut Criterion) {
+    let (m31_config, mut m31_circuit) =
+        benchmark_setup::<M31ExtConfig>(GKRScheme::GkrSquare, POSEIDON_CIRCUIT);
+    let (bn254_config, mut bn254_circuit) =
+        benchmark_setup::<BN254Config>(GKRScheme::GkrSquare, POSEIDON_CIRCUIT);
+
+    let mut group = c.benchmark_group("prove poseidon by GKR^2");
```
Comment (zhenfeizhang) [inline benches/gkr_hashes.rs:52]: "prove poseidon by GKR^2" => "single thread proving 120 poseidon by GKR^2"

**Review by zhenfeizhang — APPROVED** [review]

**Comment by tonyfloatersu** [comment]
I should leave a note here, as CI process grab keccak hashes from googleAPI, while poseidon hash circuit file is missing - I am yet unsure how the circuit file gets setup, so I leave it up to you for CI process.

**Pull request merged**